src/Drivers/CMakeLists.txt

```diff
@@ -67,9 +67,9 @@ if(HIOP_USE_MPI AND HIOP_SPARSE)
 
   if(HIOP_USE_RAJA)
       set_source_files_properties(
-	nlpPriDec_ex9_sparse_raja.cpp 
-	nlpPriDec_ex9_sparse_raja_driver.cpp 
-        PROPERTIES LANGUAGE CUDA
+        nlpPriDec_ex9_sparse_raja.cpp 
+        nlpPriDec_ex9_sparse_raja_driver.cpp 
+        $<HIOP_USE_CUDA:PROPERTIES LANGUAGE CUDA> $<HIOP_USE_HIP:PROPERTIES LANGUAGE HIP>
```

Comment:
@ashermancinelli I have problem to compile `nlpPriDec_ex9_sparse_raja` if I use `$<HIOP_USE_CUDA:PROPERTIES LANGUAGE CUDA> $<HIOP_USE_HIP:PROPERTIES LANGUAGE HIP>` on lassen, where I set `-DHIOP_USE_CUDA=ON`. Note that if I use the old commit `PROPERTIES LANGUAGE CUDA`, then everything is fine. Do I need to upgrade cmake?

Reply:
The error code I got was:
```
In file included from /g/g92/chiang7/software/hiop_lassen/src/Drivers/nlpPriDec_ex9_user_recourse_sparse_raja.hpp:13,
                 from /g/g92/chiang7/software/hiop_lassen/src/Drivers/nlpPriDec_ex9_sparse_raja.hpp:12,
                 from /g/g92/chiang7/software/hiop_lassen/src/Drivers/nlpPriDec_ex9_sparse_raja_driver.cpp:1:
/g/g92/chiang7/software/hiop_lassen/src/LinAlg/hiop_raja_defs.hpp:14:34: error: 'cuda_exec' in namespace 'RAJA' does not name a template type
   using hiop_raja_exec   = RAJA::cuda_exec<HIOP_RAJA_GPU_BLOCK_SIZE>;
                                  ^~~~~~~~~
/g/g92/chiang7/software/hiop_lassen/src/LinAlg/hiop_raja_defs.hpp:15:34: error: 'cuda_reduce' in namespace 'RAJA' does not name a type
   using hiop_raja_reduce = RAJA::cuda_reduce;
```
Note that only EX9_RAJA fails to be compiled, while EX4_RAJA can be compiled without this error.

Reply:
The error you're seeing is likely because the driver is being built with host compiler and not nvcc, which means the source file properties were messed up somewhere down the line.